lib/avrora/storage/file.ex

```diff
@@ -38,10 +38,13 @@ defmodule Avrora.Storage.File do
   """
   @impl true
   def get(key) when is_binary(key) do
-    with {:ok, body} <- read_schema_file_by_name(key),
-         do: SchemaEncoder.from_json(body, &read_schema_file_by_name/1)
+    with {:ok, schema_name} <- Name.parse(key),
+         {:ok, body} <- read_schema_file_by_name(key) do
+      SchemaEncoder.from_json(body, name: schema_name.name, reference_lookup_fun: &read_schema_file_by_name/1)
```

Comment:
We need a refactoring here, nested calls generate too much lookup table. Considering unnamed types, we need to overhaul how we store lookup tables and how we work with them.